plugins/modules/system/disk_facts.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+#
+# Copyright 2021 VMware, Inc
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: disk_facts
+version_added: "2.5.0"
+short_description: Get disk facts in the linux OS
+description:
+    - This module is to get disk facts like free space, used percentage, mount location etc in linux Ansible controller/controlled machines.
+requirements:
+  - psutil(python module)
+options:
+    name:
+        description:
+            - This is the disk name to query
```

Comment:
```suggestion
            - This is the disk name to query.
```
What's the 'name' of a disk? Do you mean something like `/dev/sda`?

Also you have default `all` that you need to document.

Reply:
yes name of the disk is something like /dev/sda. Default is to get all disk information with all options.

Reply:
The normal behavior for `_info` modules is to not have a default for such a parameter (you still have not documented `all`!), but instead list everything if the parameter is not provided (i.e. it has value `None`) and only retrieve something specific if it has been provided.